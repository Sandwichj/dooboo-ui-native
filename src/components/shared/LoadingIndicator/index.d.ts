import * as React from 'react';
import { ViewStyle, ImageSourcePropType } from 'react-native';

interface Props {
  containerStyle?: ViewStyle,
  style?: ViewStyle,
  color?: string;
  size?: number | 'small' | 'large';
  children?: any;
}

const LoadingIndicator: React.SFC<Props> = (props) => {

};

export default LoadingIndicator;
