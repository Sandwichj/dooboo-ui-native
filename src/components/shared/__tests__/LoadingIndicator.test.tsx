import 'react-native';
import * as React from 'react';
import LoadingIndicator from '../LoadingIndicator';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render, RenderResult } from '@testing-library/react-native';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[LoadingIndicator] render', () => {
  beforeEach(() => {
    props = createTestProps({ });
    component = (
      <LoadingIndicator {...props} />
    );
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON =
      renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
