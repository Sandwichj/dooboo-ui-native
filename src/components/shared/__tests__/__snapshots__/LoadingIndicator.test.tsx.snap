// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LoadingIndicator] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#969696"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
